### PR TITLE
docs: adding security & change nature checklist to PULL_REQUEST_TEMPLATE

### PR DESCRIPTION
**Pull request by brucey31**
## Context

Adding new security & nature checkboxes to the PR template used in new github repos to answer this [vanta test](https://app.vanta.com/tests/code-review-template-present-gitlab?tab=results)

[Jira ticket](url)\
[Design document](url)

<details>
<summary>To do</summary>

- [ Merge into this repo]
- [Change PR template in existing github repos]
- [Change PR template in existing gitlab repos]

</details>

### Screenshots

| Before | After |
|--------|-------|
| -      | -     |

### CODE REVIEW

**Review by alinazabara — APPROVED** [review]
👍 LGTM

~~Though the PR title change failed because it didn't follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) format, using a title like 'docs: _the rest of the description_' should work. After updating the PR title, the PR should be refreshed (e.g., by pushing an empty commit).~~ -> disregard, that's been changed

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-
-- [ ]
+## Nature of changes
 
-</details>
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation update
+- [ ] Deployment update
```
Comment (JamesPeiris) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: Looks good -  do you think there's value in aligning it with the conventional commits spec?

image

e.g.
```suggestion
- [ ] `feat:` - New feature (non-breaking change which adds functionality)
- [ ] `fix:` - Bug fix (non-breaking change which fixes an issue)
- [ ] `docs:` - Documentation update
- [ ] `ci:` - Deployment update
- [ ] `!` - Breaking change (any of the above types, but a change that would cause existing functionality to not work as expected)

Based on [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#examples)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
 
 ## Description of changes
 
 <!-- Summarize the changes you've made in this PR. -->
 
 ## QA
+- [ ] I have performed a self-review of my code
+- [ ] If it is a core feature, I have added thorough tests.
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] I have added the necessary observability elements to monitor for known edge case risks?
```
Comment (JamesPeiris) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: Nit: make it a statement
```suggestion
- [ ] I have added the necessary observability elements to monitor for known edge case risks
```

**Review by JamesPeiris — APPROVED** [review]
